src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -421,6 +421,11 @@ private async Task<WorkUnitResult> ExecuteBucket(TaskHost taskHost, ItemBucket b
                     if (requirements != null)
                     {
                         TaskLoggingContext taskLoggingContext = _targetLoggingContext.LogTaskBatchStarted(_projectFullPath, _targetChildInstance);
+                        if (taskLoggingContext.TaskName == "MSBuild")
```

Comment:
Actually, why restrict this to the MSBuild task? If the current task id of any task would be saved, the `IBuildEngine` based build requests would also get parented to their spawning tasks, right?

Reply:
Good idea, fixed. Also renamed CurrentMSBuildTask -> CurrentTaskContext.